### ISSUE TITLE
StaticFileServer tests broken when running under Xcode

### ISSUE THREAD

**Issue by djones6** [IBM-Swift/Kitura#1427]
## Context and Description
A number of Kitura's StaticFileServer tests rely on serving resources using a path of `"./Tests/KituraTests/TestStaticFileServer/"`.  This works from the command line, though only if the working directory is the root directory of the Kitura project.

If you try to run the tests from any subdirectory, such as running `swift test` from `./Tests`, then these tests will fail as the file server path will resolve to the wrong location.  This also affects Xcode, as the working directory while running tests is not the root of the project.

~I believe these tests used to pass through Xcode, so probably some of the recent tightening of the StaticFileServer has caused this to break.~
**Update**: It turns out that this is only broken _if_ the Kitura project is in a subdirectory of another SwiftPM project (for example, if it is an edited package, such as `OriginalProject/Packages/Kitura`). My guess is that there is some logic for identifying the root of the project, which is being defeated by the presence of an enclosing SPM project higher up the directory tree.

### Steps to Reproduce
Clone Kitura so that it is within another project (or edit the Kitura dependency of a Kitura-based project).
Run the Kitura tests:
1) from the command line, from the root directory of the Kitura project
2) from the command line, from a subdirectory (eg. `Kitura/Tests`)
3) through Xcode

### Expected vs. Actual Behaviour
- [ ] Expected to happen...
  - Tests pass on command line and through Xcode
- [ ] Actually happened...
  - StaticFileServer-dependent tests fail under Xcode (2 crash due to force-unwrap, being addressed in #1426), or when pwd is not at the root of the project.

**Comment by saiHemak** [comment]
@djones6  I tried to recreate this issue by following the steps mentioned in the description 
```
/Library/Developer/Toolchains/swift-4.2.3-RELEASE.xctoolchain/usr/bin/swift test
est Suite 'TestTypeSafeMiddleware' passed at 2019-03-28 14:00:00.749.
	 Executed 33 tests, with 0 failures (0 unexpected) in 3.950 (3.952) seconds
Test Suite 'KituraPackageTests.xctest' passed at 2019-03-28 14:00:00.749.
	 Executed 248 tests, with 0 failures (0 unexpected) in 19.485 (19.505) seconds
Test Suite 'All tests' passed at 2019-03-28 14:00:00.749.
	 Executed 248 tests, with 0 failures (0 unexpected) in 19.485 (19.505) seconds
```
 The tests seems to pass successfully..

**Comment by saiHemak** [comment]
I am able to reproduce the crash with the following steps :

```
1. Clone Kitura-Sample project
2. cd Kitura-Sample. swift package edit Kitura
3. cd Packages/Kitura
4.swift package generate-xcodeproj
```

**Comment by saiHemak** [comment]
On further investigation , I could observe that the  same test case passes when ran from command-line and fails when ran in Xcode .I could see the ```absolutePath ``` formed by ```StaticFileServer```  is different  in both the cases . For instance :
```absolutePath```  in  Xcode :
```Absolute path /private/tmp/./Tests/KituraTests/TestStaticFileServer
================/Users/sakandur/swift-workspace/Kitura-Sample/./Tests/KituraTests/TestStaticFileServer```

Command-line :
```
 Absolute path /Users/sakandur/swift-workspace/Kitura-Sample/Packages/Kitura/./public
================/Users/sakandur/swift-workspace/Kitura-Sample/./public
 Absolute path /Users/sakandur/swift-workspace/Kitura-Sample/Packages/Kitura/./Tests/KituraTests/TestStaticFileServer
```

As we can see ```packages``` is missing in the path formed

**Comment by saiHemak** [comment]
The ```fileManager.currentDirectoryPath```API  is used in forming the  ```absolutePath```  of the ```FileServer```. However, this  seems to be returning different paths in XCode and command line 

on Xcode ```/private/tmp``` 

commandline ``` /Users/sakandur/swift-workspace/Kitura-Sample/Packages/Kitura```
as the ```absoluteRootPath``` formed in Xcode fails ```fileManager.fileExists(atPath: absolutePath)``` check and hence ```getOriginalRepositoryPath``` has been returned .

**Comment by saiHemak** [comment]
commenting out the code which does slicing of path components till packages  in ```removePackagesDirectory``` has resolved the issue and could successfully run the test through command-line as well as through Xcode .However, we need to consider the cases

**Comment by saiHemak** [comment]
@djones6  based on the above PR and the code flow I understand that the slicing happens only when the ```pathComponents``` have ```.build or checkouts or packages ``` folder .  I believe this code path gets executed  only when the Kitura tests are triggered from a nested folder  i.e under a project .

**Comment by saiHemak** [comment]
@kilnerm, Discussed with @djones6   and assigning it to you  as we both are skeptical on  the relevance of above patch  in the context of ```StaticFileServer```.

**Comment by djones6** [comment]
TL;DR: The problem does not apply to any dependent projects, and is confined to Kitura's tests only.

Resolved by #1449 - to summarize our findings:  This is specifically a limitation when the Kitura package itself is put into edited mode, because it is the location of a Kitura source file that is used to identify the correct serving path when a relative path is given to StaticFileServer.

If I've understood correctly, what the logic is doing here is:
1. Start with the location of the `Kitura/Sources/Kitura/staticFileServer/ResourcePathHandler.swift` file (https://github.com/IBM-Swift/Kitura/blob/2.7.0/Sources/Kitura/staticFileServer/ResourcePathHandler.swift#L69)
2. Go exactly 4 directory levels up to find the directory containing the Kitura repository
3. Go enough directory levels up to find the dependent project's directory:  this is either 3 levels (if Kitura is inside `.build/checkouts/`), or 2 levels (if Kitura is in edited mode, and located within `Packages/`).

This correctly identifies the **dependent** project's root, and is what you want when you create a StaticFileServer with a relative path in your own project.  However, it's incorrect if the StaticFileServer is created from within the Kitura project itself.

However - since the only use of StaticFileServer within the Kitura package itself is from the tests, it's effectively a limitation on the tests.  #1449 replaces the relative paths with absolute ones, and thus avoids this logic altogether.
